projects/plugins/crm/src/automation/class-base-condition.php

```diff
@@ -76,4 +93,23 @@ public function condition_met(): bool {
 		return $this->condition_met;
 	}
 
+	/**
+	 * Checks if this is a valid operator for this condition and throws an
+	 * exception if the operator is invalid.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $operator The operator.
+	 * @return void
+	 *
+	 * @throws Automation_Exception If the operator is invalid for this condition.
+	 */
+	protected function check_for_valid_operator( string $operator ): void {
+		if ( ! in_array( $operator, $this->valid_operators, true ) ) {
+			$this->condition_met = false;
+			$this->logger->log( 'Invalid operator: ' . $operator );
+			throw new Automation_Exception( 'Invalid operator: ' . $operator );
```

Comment:
Good improvement in adding this generic method here. Maybe we should add a new exception code: `CONDITION_INVALID_OPERATOR`